src/core/graph.js

```diff
@@ -7,6 +7,72 @@ import {makeAddressModule, type AddressModule} from "./address";
 import {toCompat, fromCompat, type Compatible} from "../util/compat";
 import * as NullUtil from "../util/null";
 
+/**
+ * This module contains the Graph, which is one of the most fundamental pieces
+ * of SourceCred. SourceCred uses this graph to model all of the contributions
+ * that make up a project, and the relationships between those contributions.
+ *
+ * The Graph serves a simple function: it keeps track of which Nodes exist, and
+ * what Edges join those nodes to each other. Nodes and Edges are both identified
+ * by Addresses; specifically, `NodeAddressT`s and `EdgeAddressT`s.
+ *
+ * In both cases, addresses are modeled as arrays of strings. For example,
+ * you might want to give an address to your favorite node. You can do so as
+ * follows:
+ *
+ * const myAddress: NodeAddressT = NodeAddress.fromParts(["my", "favorite"])
+ *
+ * Edge Addresses are quite similar, except you use the EdgeAddress module.
+ *
+ * We model addresses as arrays of strings so that plugins can apply
+ * hierarchical namespacing for the address. In general, for any address, the
+ * first piece should be the name of the organization that owns the plugin, and
+ * the second piece should be the name of the plugin. Pieces thereafter are
+ * namespaced by the plugin's internal logic. For example, SourceCred has a
+ * Git plugin, and that plugin produces addresses like
+ * ["sourcecred", "git", "commit", "9cba0e9e212a287ce26e8d7c2d273e1025c9f9bf"].
+ *
+ * This enables "prefix matching" for finding only certain types of nodes. For
+ * example, if we wanted to find every Git commit in the graph, we
+ * could use the following code:
+ *
+ * const commitPrefix = NodeAddress.fromParts(["sourcecred", "git", "commit"]);
+ * const commitNodes = `graph.nodes({prefix: commitPrefix})`;
```

Comment:
Is `graph.nodes...` meant to be inside a template literal?

Reply:
Good catch; it shouldn’t be.

(Aside: In Rust, if you put code into a doc comment, it’s automatically
turned into a test case. Would be nice to have something similar here.)